src/routes/consumers/detail.$username/credentials/detail.$id.tsx

```diff
@@ -0,0 +1,127 @@
+import { useBoolean } from 'react-use';
+import { useEffect } from 'react';
+import { APISIX } from '@/types/schema/apisix';
+import {
+  getCredentialQueryOptions,
+  putCredentialReq,
+} from '@/apis/credentials';
+import { FormPartCredential } from '@/components/form-slice/FormPartCredential';
+import { pipeProduce } from '@/utils/producer';
+import { DetailCredentialsTabs } from '@/components/page-slice/consumers/DetailCredentialsTabs';
+
+type CredentialFormProps = {
+  readOnly: boolean;
+  setReadOnly: (v: boolean) => void;
+};
+
+const CredentialDetailForm = (props: CredentialFormProps) => {
+  const { readOnly, setReadOnly } = props;
+  const { t } = useTranslation();
+  const { username, id } = useParams({
+    from: '/consumers/detail/$username/credentials/detail/$id',
+  });
+
+  const {
+    data: credentialData,
+    isLoading,
+    refetch,
+  } = useSuspenseQuery(getCredentialQueryOptions(username, id));
+
+  const form = useForm({
+    resolver: zodResolver(APISIX.CredentialPut),
+    shouldUnregister: true,
+    shouldFocusError: true,
+    mode: 'all',
+    disabled: readOnly,
+  });
+
+  useEffect(() => {
+    if (credentialData?.value && !isLoading) {
+      form.reset(credentialData.value);
+    }
+  }, [credentialData, form, isLoading]);
+
+  const putCredential = useMutation({
+    mutationFn: putCredentialReq,
+    async onSuccess() {
+      notifications.show({
+        message: t('consumers.credentials.edit.success'),
+        color: 'green',
+      });
+      await refetch();
+      setReadOnly(true);
+    },
+  });
+
+  if (isLoading) {
+    return <Skeleton height={400} />;
+  }
+
+  return (
+    <FormProvider {...form}>
+      <form
+        onSubmit={form.handleSubmit((d) => {
+          putCredential.mutateAsync({
+            username,
+            ...pipeProduce()(d),
+          });
+        })}
+      >
+        <FormPartCredential showDate />
+        {!readOnly && (
+          <Group>
+            <FormSubmitBtn>{t('form.btn.save')}</FormSubmitBtn>
+            <Button variant="outline" onClick={() => setReadOnly(true)}>
+              {t('form.btn.cancel')}
+            </Button>
+          </Group>
+        )}
+      </form>
+    </FormProvider>
+  );
+};
+
+function RouteComponent() {
+  const { t } = useTranslation();
+  const [readOnly, setReadOnly] = useBoolean(true);
+
+  return (
+    <>
+      <DetailCredentialsTabs />
+      <PageHeader
+        title={t('consumers.credentials.edit.title')}
+        {...(readOnly && {
+          title: t('consumers.credentials.detail.title'),
+          extra: (
+            <Button
+              onClick={() => setReadOnly(false)}
+              size="compact-sm"
+              variant="gradient"
+            >
+              {t('form.btn.edit')}
+            </Button>
+          ),
+        })}
+      />
+      <FormTOCBox>
+        <CredentialDetailForm readOnly={readOnly} setReadOnly={setReadOnly} />
+      </FormTOCBox>
+    </>
```

Comment:
Is there a button here to return to the credential list?

Reply:
Users can directly click on the Credentials tab at the top